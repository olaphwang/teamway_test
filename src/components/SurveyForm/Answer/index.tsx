import React from 'react';
import styles from './index.scss';

interface AnswerProps {
  answerId: string;
  answer: string;
  onInputChange: (index: string) => void;
  checked: boolean;
}
export const Answer = ({ answerId, answer, onInputChange, checked }: AnswerProps) => {
  return (
    <div className={styles.answerWrapper}>
      <input
        type="radio"
        name="answer"
        id={answerId}
        onChange={() => {
          onInputChange(answer);
        }}
        checked={checked}
      />
      <label id={answerId} htmlFor={answerId} className={styles.answerWord}>
        {answer}
      </label>
    </div>
  );
};

export default Answer;
