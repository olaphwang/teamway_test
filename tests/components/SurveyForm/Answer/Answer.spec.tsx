import React from 'react';
import renderer from 'react-test-renderer';
import Answer from '@components/SurveyForm/Answer';

describe('Answer component', () => {
  test('should render component properly', () => {
    // when
    const componentRenderer = renderer.create(<Answer answerId="1" checked={true} answer="11" onInputChange={null} />);
    const tree = componentRenderer.toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
