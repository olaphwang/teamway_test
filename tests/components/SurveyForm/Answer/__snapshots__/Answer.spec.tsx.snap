// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer component should render component properly 1`] = `
<div
  className="answerWrapper"
>
  <input
    checked={true}
    id="1"
    name="answer"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="answerWord"
    htmlFor="1"
    id="1"
  >
    11
  </label>
</div>
`;
